### ISSUE TITLE
Need Help to customize "CustomMarkDownSummary"

### ISSUE THREAD

**Issue by parimalm** [rfennell/vNextBuild#175]
### VSTS Extension you are using
StyleCop Runner Build Task

### Where are you running it?
VSTS

### Version of Extension/Task
Current Public Version

### Expected behavior and actual behavior
I’m using Stylecop Runner Build Task and I want to customize build report. The current build report includes number of violations across projects , instead of this I want to include detailed report (i.e. what actually went wrong), with details, similar to what we see in Visual Studio when we scan our code with Stylecop. 

CustomMarkDownSummary-stylecopsummary.md
StyleCop
•	Project [Stylecop] - [7] Violations
StyleCop found [7] violations across [1] projects

### Steps to reproduce the problem
1. Add Stylecop runner as build task.
2. Commit code having violation count more than specified "Max. Violation Count".

**Comment by ChrisLGardner** [comment]
I think this might be possible, we should be able to capture the stylecop output and return that to the console/build log. 

I'll have to investigate this and do some testing, it might be a week or two before I have time however.

**Comment by parimalm** [comment]
Thanks a lot Chris. looking forward to hear from you.

**Comment by parimalm** [comment]
Hi Chris,
Sorry to bother you again, just wanted to check if you've had time to look into it.

**Comment by ChrisLGardner** [comment]
Sorry, it's been pretty hectic recently. I'll get to this as soon as I can.

**Comment by rfennell** [comment]
You do get the StyleCop output as generated by tool itself (i.e not the VSTS task) as both a text file and an XML file. As  remember this available as an artifact  

As everyone's need for custom reporting layout is different it might be easier to use inline powershell to process either of these files to the format required.

**Comment by parimalm** [comment]
Thanks @rfennell , but I need to present and store output of build tasks. Every individual can change the ruleset on local dev box and change the report accordingly.

**Comment by rfennell** [comment]
The build summary seems the wrong place for a potentially very long StyleCop issue list to me. Maybe your legacy code base is cleaner. 

Could I suggest a middle ground of adding some text to the build summary that says something like  _'For full details of the Stylecop issues see the detailed log file [link to artifact text file]'_ 

That said the logic that write the build summary 

```
Add-Content $summaryMdPath "###StyleCop"
Add-Content $summaryMdPath ($result.Summary)
Add-Content $summaryMdPath ("`nStyleCop found [{0}] violations across [{1}] projects" -f $result.TotalViolations, $result.ProjectsScanned)

```

could easily have a for loop added after it, probably guarded by an advanced settings to enable and disable it, that lists all the issues in a bit more detail. So maybe that is the better solution for your needs

I should also inform you that there is currently active work in progress to enable StyleCop 5 support, the current public task is StyleCop 4.7 based. We a a problem with some dictionary based rules at present, but I want to get this release out before starting other work on this task.

**Comment by rfennell** [comment]
I could not see an easy way to introduce a template, but I have added a basic choice of simple summary (the old way and still the default) and detailed summary that list the filenames and violations. Hope this is enough.

This has been released as 2.1.1
